### PR TITLE
Feature/model namespacing

### PR DESCRIPTION
**Pull request by JamieCee20**
**What has changed?**

Update to the MakeResourceCommand file that will allow the argument of namespace in the scenarios where generating a resource for a model isn't in the App\Models path.

**Usage**

> make:filament-resource {name?} {--namespace=} {--soft-deletes} {--view} {--G|generate} {--S|simple} {--panel=} {--F|force}

Pass the namespace argument to detect the model in that namespace, for example --namespace=My\Custom\Namespace

### CODE REVIEW

**Comment by danharrin** [comment]
I think it should be `--namespace` instead of `--model`

Please remove the `dd()` and other formatting changes that your IDE made

**Comment by JamieCee20** [comment]
> I think it should be `--namespace` instead of `--model`
> 
> Please remove the `dd()` and other formatting changes that your IDE made

I shall do that tomorrow for the dd, apologies as well, I have an autoformatter following psr-2, Ill turn it off haha.

Could I just get some clarification on the --namespace not --model as Im sure I set it as --namespace?

**Comment by danharrin** [comment]
Sorry, I meant it should be `--model` instead of `--namespace`. My fault

**Comment by JamieCee20** [comment]
@danharrin I have made the changes to use --model instead of --namespace, as well as changing the stub to prevent always using App\Models before including the model namespace/class.

File: packages/panels/src/Commands/Aliases/MakeResourceCommand.php [hunk 1]
```diff
@@ -8,5 +8,5 @@ class MakeResourceCommand extends Commands\MakeResourceCommand
 {
     protected $hidden = true;
 
-    protected $signature = 'filament:resource {name?} {--soft-deletes} {--view} {--G|generate} {--S|simple} {--panel=} {--F|force}';
+    protected $signature = 'filament-resource {name?} {--model=} {--soft-deletes} {--view} {--G|generate} {--S|simple} {--panel=} {--F|force}';
```
Comment (zepfietje) [inline packages/panels/src/Commands/Aliases/MakeResourceCommand.php]: Seems like an unintentional breaking change?

```suggestion
    protected $signature = 'filament:resource {name?} {--model=} {--soft-deletes} {--view} {--G|generate} {--S|simple} {--panel=} {--F|force}';
```

**Comment by danharrin** [comment]
Hi, thanks for the PR. I think it's important to give you a bit of information as to why this PR has been tagged as "low priority". We really appreciate all contributions, and intend to visit this in the future, and hopefully merge it if it is a suitable feature for the framework.

However, reviewing PRs, and often rewriting them on behalf of the author to ensure they are consistent with the rest of the framework, consumes a lot of my time. At the moment we have a large backlog of bug reports that I am prioritising, since most of these affect functionality that worked in a previous version and are now impacting real projects. Hopefully you can see how that work comes first before introducing new features that people aren't using in projects yet.

Again, thank you for contributing and I'm looking forward to reviewing this PR when our backlog of bug reports has reduced.

File: packages/panels/src/Commands/MakeResourceCommand.php [hunk 2]
```diff
@@ -271,4 +259,26 @@ public function handle(): int
 
         return static::SUCCESS;
     }
-}
+    
+    protected function getModelFromNamespace($modelName)
```
Comment (danharrin) [inline packages/panels/src/Commands/MakeResourceCommand.php]:
```suggestion
    protected function getModelFromNamespace($modelName): string
```

File: packages/panels/src/Commands/MakeResourceCommand.php [hunk 3]
```diff
@@ -191,9 +179,9 @@ public function handle(): int
         $this->copyStubToApp('Resource', $resourcePath, [
             'eloquentQuery' => $this->indentString($eloquentQuery, 1),
             'formSchema' => $this->indentString($this->option('generate') ? $this->getResourceFormSchema(
-                'App\\Models' . ($modelNamespace !== '' ? "\\{$modelNamespace}" : '') . '\\' . $modelClass,
+                ($modelNamespace !== '' ? "\\{$modelNamespace}" : 'App\\Models') . '\\' . $modelClass,
             ) : '//', 4),
-            'model' => $model === 'Resource' ? 'Resource as ResourceModel' : $model,
+            'model' => $model === 'Resource' ? 'Resource as ResourceModel' : (($this->option('model')) ? $model : "App\Models\\" . $model),
```
Comment (danharrin) [inline packages/panels/src/Commands/MakeResourceCommand.php]:
```suggestion
            'model' => $model === 'Resource' ? 'Resource as ResourceModel' : ($this->option('model') ? $model : 'App\Models\\' . $model),
```

File: packages/panels/src/Commands/MakeResourceCommand.php [hunk 4]
```diff
@@ -24,26 +24,14 @@ class MakeResourceCommand extends Command
...
-            ->studly()
-            ->replace('/', '\\');
-
-        if (blank($model)) {
-            $model = 'Resource';
-        }
+        $modelName = $this->argument('name');
+        $modelNamespace = $this->option('model');
+        
+        $model = $modelNamespace ? $this->getModelFromNamespace($modelNamespace) : $this->getModelFromNamespace($modelName);
```
Comment (danharrin) [inline packages/panels/src/Commands/MakeResourceCommand.php]:
```suggestion
        $model = $this->getModelFromNamespace(filled($modelNamespace) ? $modelNamespace : $modelName);
```

**Review by danharrin — CHANGES_REQUESTED** [review]

File: packages/panels/src/Commands/MakeResourceCommand.php [hunk 5]
```diff
@@ -191,9 +179,9 @@ public function handle(): int
         $this->copyStubToApp('Resource', $resourcePath, [
             'eloquentQuery' => $this->indentString($eloquentQuery, 1),
             'formSchema' => $this->indentString($this->option('generate') ? $this->getResourceFormSchema(
-                'App\\Models' . ($modelNamespace !== '' ? "\\{$modelNamespace}" : '') . '\\' . $modelClass,
+                ($modelNamespace !== '' ? "\\{$modelNamespace}" : 'App\\Models') . '\\' . $modelClass,
             ) : '//', 4),
-            'model' => $model === 'Resource' ? 'Resource as ResourceModel' : $model,
+            'model' => $model === 'Resource' ? 'Resource as ResourceModel' : ($this->option('model') ? $model : 'App\Models\\' . $model),
```
Comment (danharrin) [inline packages/panels/src/Commands/MakeResourceCommand.php]:
```suggestion
            'model' => ($model === 'Resource') ? 'App\\Models\\Resource as ResourceModel' : ($this->option('model') ? $model : ('App\\Models\\' . $model)),
```

**Review by danharrin — APPROVED** [review]

**Pull request merged**